yaml_edit/lib/src/editor.dart

```diff
@@ -417,17 +407,15 @@ class YamlEditor {
   /// - 2 # comment 2
   /// '''
   /// ```
-  YamlNode remove(Iterable<Object> path) {
-    ArgumentError.checkNotNull(path, 'path');
-
-    SourceEdit edit;
-    YamlNode expectedNode;
+  YamlNode remove(Iterable<Object?> path) {
+    SourceEdit? edit;
+    var expectedNode = wrapAsYamlNode(null);
```

Comment:
Setting a default `expectedNode` here. I don't know if this is the right way to do it.

Reply:
This is fine... but this is only necessary when in the `if (path.isEmpty) {` case...

So we could make `late expectedNode` and only do `expectedNode = wrapAsYamlNode(null);` in the case where `path.isEmpty`.